### ISSUE TITLE
Problem with SELECT DISTINCT in partition key

### ISSUE THREAD

**Issue by vanessaschissato** [Impetus/kundera#591]
Hi,

I'm trying to make a 'select distinct' in the partition key (my entity has a composite key), but i'm receiving an error (below).
CQL 3 allow this: https://issues.apache.org/jira/browse/CASSANDRA-4536

Entity 'Tag':

@Entity
@Table(name = "tags", schema = "a")
public class Tag {

```
@EmbeddedId
private TagId id;

// setters e getters
```

}

Entity 'TagId':

@Embeddable
public class TagId
{
    @Column(name = "keya") // <----------- PARTITION KEY
    private String keya;

```
@Column(name = "keyb")
private String keyb;

@Column(name = "keyc")
private String keyc;

    // setters e getters
```

}

My query:

em.createQuery("SELECT DISTINCT(t.id.keya) FROM Tag t").getResultList();

And this is the stack trace:

Caused by: java.lang.IllegalArgumentException: attribute of the given name and type is not present in the managed type, for name:id.keya)
    at com.impetus.kundera.metadata.model.type.AbstractManagedType.checkForValid(AbstractManagedType.java:1032)
    at com.impetus.kundera.metadata.model.type.AbstractManagedType.getAttribute(AbstractManagedType.java:523)
    at com.impetus.client.cassandra.query.CassQuery.getColumnList(CassQuery.java:342)
    at com.impetus.client.cassandra.query.CassQuery.onQueryOverCQL3(CassQuery.java:645)
    at com.impetus.client.cassandra.query.CassQuery.populateEntities(CassQuery.java:136)
    at com.impetus.kundera.query.QueryImpl.fetch(QueryImpl.java:937)
    at com.impetus.kundera.query.QueryImpl.getResultList(QueryImpl.java:161)
    at ...

**Comment by kkmishra** [comment]
Hi,

Selecting the entire composite key will work for you, however selecting partial attribute of composite key is not yet supported and also DISTINCT keyword.

Following query should work for you.

```
em.createQuery("SELECT t.id FROM Tag t").getResultList();
```

It is a bug, will provide a fix soon.

-kuldeep

**Comment by kkmishra** [comment]
Hi,

Selecting partial attribute of composite key is only requirement or you want to select the partial attribute of another embedded objects as well ?

Partial key selection on composite id would be released with 2.12, but support for later one you have to wait for > 2.12 releases

Please provide your input.

-Kuldeep

**Comment by vanessaschissato** [comment]
My actual need is only selecting partial attribute of composite keys.

**Comment by vanessaschissato** [comment]
I'm using Kundera 2.11.1 and i'll be waiting for the 2.12 ;)

For now i'm using a native query that works for me.

**Comment by chhavigangwal** [comment]
Issue fixed. Released with Kundera-2.12.
